Lib/fontTools/varLib/models.py

```diff
@@ -61,13 +61,16 @@ def normalizeValue(v, triple, extrapolate=False):
         )
     if not extrapolate:
         v = max(min(v, upper), lower)
-    if v == default:
-        v = 0.0
-    elif v < default:
-        v = (v - default) / (default - lower)
-    else:
-        v = (v - default) / (upper - default)
-    return v
+
+    if v == default or lower == upper:
+        return 0.0
+
+    if (v < default and lower != default) or (v > default and upper == default):
```

Comment:
I'm not sure about the logic change. If  `default == upper`, I don't think we should extrapolate using `lower`.

Reply:
hm.. then I misunderstood the intent of this extrapolate=True.. Isn't this what we want?

```python
In [1]: from fontTools.varLib.models import *

In [2]: normalizeLocation({"wght": 550}, {"wght": (100, 400, 400)}, extrapolate=False)  # clamps to 0.0
Out[2]: {'wght': 0.0}

In [3]: normalizeLocation({"wght": 550}, {"wght": (100, 400, 400)}, extrapolate=True)  # extends beyond
Out[3]: {'wght': 0.5}
```

Reply:
note that normalizeLocation is currently never called with extrapolate=True, the VariationModel already takes locations that are assumed to be normalized (i.e. 0 as default).
Only your PR branch for instancing-L4 calls normalizeValue (which this normalizeLocation calls upon) with extrapolaate=True, in the rebaseTent method.. 

I think @simoncozens needs normalizeLocation(extrapolate=True) for his Smart Components glyphsLib PR, see https://github.com/googlefonts/glyphsLib/pull/822/files#diff-63af9ecacc6b3c6fa71c485ef068d441e09bbb7a7f936291eaa700e18d1611c9R119

Reply:
IMO what Cosimo shows is exactly what we want. You normalize using the axis min/default/max, but don't clamp.

Reply:
so, have we reached a consensus? 
@simoncozens have you tried seeing if using this proposed `normalizeLocation(extrapolate=True)` does what you want in your glyphsLib PR? It's the main reason we are adding this

Reply:
Yes, it works.

Reply:
ok thanks. Then I think we can merge this since it's useful. If it's desirable to not do that, but instead raise ValueError as behdad suggested, we could add that as an option later. I personally can't think of a situation where that would apply considering that extrapolate is already opt-in and disabled by default.
By the way, the `instancer-l4` branch tests continue to pass after I locally merge in the current `fix-division-by-zero` PR branch, so the latter changes don't seem to affect the new instancer.solver.